packages/common/src/grpc-service.js

```diff
@@ -351,6 +358,13 @@ GrpcService.prototype.requestStream = function(protoOpts, reqOpts) {
     shouldRetryFn: GrpcService.shouldRetryRequest_,
 
     request: function() {
+      try {
+        reqOpts = util.decorateRequest(reqOpts, { projectId: self.projectId });
+      } catch (e) {
+        stream.destroy(e);
```

Comment:
Should we be destroying within a `setImmediate` and returning the stream here like in https://github.com/GoogleCloudPlatform/google-cloud-node/pull/1656/files#diff-78d3ba232fe7028b5068f61d6d20a988R439?